homeassistant/components/esphome/sensor.py

```diff
@@ -74,6 +78,52 @@ async def async_setup_entry(
 class EsphomeSensor(EsphomeEntity[SensorInfo, SensorState], SensorEntity):
     """A sensor implementation for esphome."""
 
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        await super().async_added_to_hass()
+
+        if self._static_info.last_reset_type is LastResetType.NEVER:
+            self._attr_last_reset = datetime.min
+        elif self._static_info.last_reset_type is LastResetType.AUTO:
+
+            self.async_on_remove(
+                async_track_state_change_event(
```

Comment:
Don't track state, but instead use a method override on the update callback. That update callback currently does not exist, so add that:

- Add method `_data_updated` to `EsphomeEntity`
  ```python
  @callback
  def _state_updated(self) -> None:
      self.async_write_ha_state()
  ```
- If `last_reset_type` is set to auto, override it:
  ```python
  _old_state = None

  @callback
  def _state_updated(self) -> None:
      if self._old_state is not None:
          # Figure out last reset          
          ...

      self._old_state = self._state
      super()._state_updated()
  ```